consensus/state_processing/src/per_block_processing/process_operations.rs

```diff
@@ -658,11 +675,73 @@ pub fn process_consolidation_requests<E: EthSpec>(
     Ok(())
 }
 
+fn is_valid_switch_to_compounding_request<E: EthSpec>(
+    state: &BeaconState<E>,
+    consolidation_request: &ConsolidationRequest,
+    spec: &ChainSpec,
+) -> Result<bool, Error> {
+    // Switch to compounding requires source and target be equal
+    if consolidation_request.source_pubkey != consolidation_request.target_pubkey {
+        return Ok(false);
+    }
+
+    // Verify pubkey exists
+    let Some(source_index) = state
+        .pubkey_cache()
+        .get(&consolidation_request.source_pubkey)
+    else {
+        // source validator doesn't exist
+        return Ok(false);
+    };
+
+    let source_validator = state.get_validator(source_index)?;
```

Comment:
This last line is the only reason this function returns a `Result<bool>` rather than a `bool`. But it shouldn't ever return an error if the line above returned a pubkey. But since you need to fetch this validator here anyway, just combine them so this function has no reliance on the pubkey cache:
```rust
let source_validator = match state.get_validator(source_index) {
    Ok(v) => v,
    Err(_) => return false, 
};
```
then you can refactor this whole function to just return a `bool`

Reply:
Resolved in [9f85074](https://github.com/sigp/lighthouse/pull/6496/commits/9f850748548612f004ff009dc3e8fce7d6829898)